### ISSUE TITLE
Screen glitches after plugging in to external screen/Docking station 

### ISSUE THREAD

**Issue by leosunmo** [regolith-linux/regolith-desktop#133]
In Regolith Unstable I am currently getting screen issues after unplugging and plugging back in to my ThinkPad docking station.

![image](https://user-images.githubusercontent.com/13341934/65560820-3478da00-df94-11e9-93a4-ad3a8ff923fc.png)

As you can see it draws the windows incorrectly and the black areas are flashing.

After unplugging and plugging it back in after that it gets worse. It just displays the default background color (gray for Nordic theme) with a sliver of my background picture across all displays. I am able to log in and can see my mouse change shape but the screen does not update. Only fix is forceful shutdown and restart.

**Comment by leosunmo** [comment]
Turning off Compton seems to have fixed this issue, so will have to investigate further down the Compton route.

**Comment by kgilmer** [comment]
May be related: https://github.com/regolith-linux/regolith-desktop/issues/116

**Comment by waltsims** [comment]
After updating to regolith 2.30 I can also not use external monitors (both HDMI and Display Port)

```
regolith-assets/disco,now 1.09-1ubuntu1~ppa1 amd64 [installed,automatic]
  Desktop assets for Regolith Linux

regolith-compton-config/disco,now 1.5.2-1ubuntu1~ppa1 amd64 [installed,automatic]
  Regolith configuration for Compton

regolith-conky-config/disco,now 2.01-1ubuntu1~ppa1 amd64 [installed,automatic]
  Regolith configuration for Conky

regolith-desktop/disco,now 2.30-1ubuntu1~ppa1 amd64 [installed]
  Regolith Desktop

regolith-gdm3-theme/disco 1.04-1ubuntu1~ppa1 amd64
  Regolith gdm3 based on Ubuntu 18.04 theme.

regolith-gnome-flashback/disco,now 1.46-1ubuntu1~ppa1 amd64 [installed]
  Regolith customized gnome-flashback session

regolith-i3-gaps-config/disco,now 1.18.19-1ubuntu1~ppa1 amd64 [installed,automatic]
  Regolith i3-gaps config file

regolith-i3-wm/disco 4.16-1ubuntu18ppa14 amd64
  improved dynamic tiling window manager

regolith-i3blocks/disco 1.4-1ubuntu9ppa1 amd64
  highly flexible status line for the i3-gaps window manager

regolith-i3xrocks-config/disco,now 1.14-1ubuntu1~ppa1 amd64 [installed,automatic]
  Regolith configuration for i3xrocks launcher

regolith-rofi-config/disco,now 1.2-1ubuntu1~ppa1 amd64 [installed,automatic]
  Regolith configuration for rofi launcher

regolith-scripts/disco 1.8-1ubuntu1ppa1 amd64
  Scripts for Regolith Linux

regolith-st/disco,now 0.8.2-1ubuntu20ppa2 amd64 [installed,automatic]
  simple terminal emulator for X

regolith-styles/disco,now 1.2.35-1ubuntu1~ppa1 amd64 [installed,automatic]
  Theme and style definitions for Regolith Linux

regolith-xeventbind/disco,now 1.3-1ubuntu1~ppa1 amd64 [installed,automatic]
  Configuration to refresh desktop wallpaper after screen changes.

```


```
            .-/+oossssoo+/-.               walter@x1carbon
        `:+ssssssssssssssssss+:`           ---------------
      -+ssssssssssssssssssyyssss+-         OS: Ubuntu 19.04 x86_64
    .ossssssssssssssssssdMMMNysssso.       Host: 20KGS59A00 ThinkPad X1 Carbon 6th
   /ssssssssssshdmmNNmmyNMMMMhssssss/      Kernel: 5.0.0-29-generic
  +ssssssssshmydMMMMMMMNddddyssssssss+     Uptime: 1 hour, 16 mins
 /sssssssshNMMMyhhyyyyhmNMMMNhssssssss/    Packages: 2321 (dpkg), 14 (snap)
.ssssssssdMMMNhsssssssssshNMMMdssssssss.   Shell: zsh 5.5.1
+sssshhhyNMMNyssssssssssssyNMMMysssssss+   Resolution: 2560x1440
ossyNMMMNyMMhsssssssssssssshmmmhssssssso   DE: GNOME 3.32.2
ossyNMMMNyMMhsssssssssssssshmmmhssssssso   WM: i3
+sssshhhyNMMNyssssssssssssyNMMMysssssss+   Theme: Nordic [GTK2/3]
.ssssssssdMMMNhsssssssssshNMMMdssssssss.   Icons: Paper [GTK2/3]
 /sssssssshNMMMyhhyyyyhdNMMMNhssssssss/    Terminal: st
  +sssssssssdmydMMMMMMMMddddyssssssss+     Terminal Font: Liberation Mono
   /ssssssssssshdmNNNNmyNMMMMhssssss/      CPU: Intel i7-8650U (8) @ 4.200GHz
    .ossssssssssssssssssdMMMNysssso.       GPU: Intel UHD Graphics 620
      -+sssssssssssssssssyyyssss+-         Memory: 3181MiB / 15785MiB
        `:+ssssssssssssssssss+:`
            .-/+oossssoo+/-.

```

I have tried turning off compton (by commenting out the line in the i3 configuration) but that hasn't seemed to help.

**Comment by SteadBytes** [comment]
I am also experiencing this on a Lenovo Thinkpad X1 Carbon 7th gen; plugging in an external display often causes the display manager to seemingly crash completely with the affects @leosunmo has described. There are also issues with the HiDPI display scaling but that's a separate issue :sweat: 

Let me know if there's anything I can do to help diagnose and/or fix the issue :+1:

**Comment by waltsims** [comment]
I updated the firmware in the gnome software center and that fixed my external monitor issues. Something to try.

**Comment by SteadBytes** [comment]
Thanks @waltsims, all firmware is up to date for me unfortunately :thinking:

**Comment by leosunmo** [comment]
As Ken commented in https://github.com/regolith-linux/regolith-desktop/issues/63#issuecomment-549069627, have you tried disabling Compton? Here's a quick how-to on killing Compton: https://github.com/regolith-linux/regolith-desktop/wiki/HowTo:-Disable-Compton.

**Comment by SteadBytes** [comment]
In response to @kgilmer comment https://github.com/regolith-linux/regolith-desktop/issues/63#issuecomment-549069627 . Plugging in an external monitor causes the resolution of the laptop display to be set to 1280x720 and both displays to glitch (see images) as well as flicker to black for 5-10 seconds. Sometimes opening the gnome display settings causes it to stop and allows the resolution to be set correctly again.
![IMG_20191105_081046](https://user-images.githubusercontent.com/14834132/68455209-b2b2d800-01f2-11ea-83bc-608589340edd.jpg)
![IMG_20191105_081055](https://user-images.githubusercontent.com/14834132/68455213-b5153200-01f2-11ea-835f-bf447a411301.jpg)

**Comment by kgilmer** [comment]
Apologies if I've already asked you this, but does the problem persist after removing Compton?  

https://github.com/regolith-linux/regolith-desktop/wiki/HowTo:-Disable-Compton

**Comment by SteadBytes** [comment]
Yes it still persists for me after disabling Compton, sorry I didn't
specify that 😊

**Comment by kgilmer** [comment]
Hmm, ok.  I'm not sure what else to try off hand.  I'll update the ticket if something comes to mind.  FWIW I do not have any issues with external monitors and use gnome-control-center to manage them.

**Comment by mstr3336** [comment]
Just to add to this, disabling Compton mostly worked in my setup.
I will also note that, for my setup, I have four monitors and an NVIDIA GPU, and configured my monitor arrangement with the NVIDIA config tool. 

On Ubuntu 18.04, long before I installed regolith, I did have a lot of problems making nvidia's x configuration (it's "metamode" settings for the four monitors) persist between reboots, and I wound up writing a bash script that I added to startup/login that correctly set the correct metamode configuration.

Depending on whether this is a problem that you seem to get on the standard Ubuntu DE, or if it's unique to regolith, maybe an approach like this might help.

In my case regolith was installed from package on my 18.04 system (with the NVIDIA config login script still active) and works like a charm after disabling Compton.
